### PR TITLE
Networking Cleanup

### PR DESCRIPTION
**Pull request by AnthonyMDev**
## Changelog Additions
- **Breaking:** Renamed some of the request chain interceptors object:
  - `LegacyInterceptorProvider` -> `DefaultInterceptorProvider`
  - `LegacyCacheReadInterceptor` -> `CacheReadInterceptor`
  - `LegacyCacheWriteInterceptor` -> `CacheWriteInterceptor`
  - `LegacyParsingInterceptor` -> `JSONResponseParsingInterceptor`
- **Breaking:** `WebSocketTransport` is now initialized with an `ApolloWebSocket` (or other object conforming to the `ApolloWebSocketClient` protocol.) Previously, the initializer took in the necessary parameters to create the web socket internally. This provides better dependency injection capabilities and makes testing easier.
- Removed class constraint on `ApolloInterceptor` and converted to structs for all interceptors that could be structs instead of classes.

## Note: 
Before releasing, updates to docs need to go out. @AnthonyMDev has a stashed changeset where he started this. (Finish by running script generation and removing all instances of “Legacy”, "FlexibleDecoder", and "Parseable" left in docs!)

### CODE REVIEW

**Comment by designatednerd** [comment]
For the curious: This came out of an internal discussion about a bunch of stuff that was intended under prior plans that we've changed direction on, so we're renaming things to be more accurate for our current direction.

File: Tests/ApolloTests/WebSocket/WebSocketTransportTests.swift [hunk 1]
```diff
@@ -7,17 +7,6 @@ import Starscream
 class WebSocketTransportTests: XCTestCase {
 
   private var webSocketTransport: WebSocketTransport!
-  static var webSocketProvider: ApolloWebSocketClient.Type = MockWebSocket.self
-
-  override class func setUp() {
-    super.setUp()
-    swap(&webSocketProvider, &WebSocketTransport.provider)
```
Comment (calvincestari) [inline Tests/ApolloTests/WebSocket/WebSocketTransportTests.swift:14]: 👍🏻

File: Sources/Apollo/JSONResponseParsingInterceptor.swift [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    case noResponseToParse
+    case couldNotParseToLegacyJSON(data: Data)
+    
+    public var errorDescription: String? {
+      switch self {
+      case .noResponseToParse:
+        return "The Codable Parsing Interceptor was called before a response was received to be parsed. Double-check the order of your interceptors."
+      case .couldNotParseToLegacyJSON(let data):
+        var errorStrings = [String]()
+        errorStrings.append("Could not parse data to legacy JSON format.")
```
[the remark this answers is not in the document]
Comment (AnthonyMDev) [reply Sources/Apollo/JSONResponseParsingInterceptor.swift]: Thanks fixed!

File: Sources/Apollo/DefaultInterceptorProvider.swift [hunk 3]
```diff
@@ -0,0 +1,45 @@
+import Foundation
+
+/// The default interceptor provider for typescript-generated code
+open class DefaultInterceptorProvider: InterceptorProvider {
```
[the remark this answers is not in the document]
Comment (AnthonyMDev) [reply Sources/Apollo/DefaultInterceptorProvider.swift:4]: Yeah, I think that's overkill since we're not at 1.0 yet.

**Review by calvincestari — APPROVED** [review]

**Pull request merged**